src/js/module.js

```diff
@@ -262,6 +269,35 @@ iotjs_module_t.load = function(id, parent) {
 };
 
 
+function _makeRequireFunction(mod) {
+  var Module = mod.constructor;
+  function require(id) {
+    return mod.require(id);
+  }
+
+  function _resolve(request) {
```

Comment:
Could `_resolve` be moved out of local scope?

Reply:
But it is not necessary I think,
node/lib/internal/modules/cjs/helpers.js: 19
```js
function require(path) {
    try {
      exports.requireDepth += 1;
      return mod.require(path);
    } finally {
      exports.requireDepth -= 1;
    }
  }

  function resolve(request, options) {
    validateString(request, 'request');
    return Module._resolveFilename(request, mod, false, options);
  }

  require.resolve = resolve;

```
I consult node does.

Reply:
Yes, it's private. I mean the function could be moved out from scope makeRequireFunction.

Reply:
Why is this necessary? And node do this too.